tests/python/pants_test/backend/python/interpreter_selection_utils.py

```diff
@@ -36,12 +49,22 @@ def python_interpreter_path(version):
     command = ['python{}'.format(version), '-c', 'import sys; print(sys.executable)']
     py_path = subprocess.check_output(command).decode('utf-8').strip()
     return os.path.realpath(py_path)
-  except subprocess.CalledProcessError:
+  except (subprocess.CalledProcessError, FileNotFoundError):
     return None
 
 
+def skip_unless_any_pythons_present(*versions):
+  """A decorator that only runs the decorated test method if any of the specified pythons are present.
+
+  :param string *versions: Python version strings, such as 2.7, 3.
+  """
+  if any(v for v in versions if has_python_version(v)):
+    return skipIf(False, 'At least one of the expected python versions found.')
+  return skipIf(True, 'Could not find at least one of the required pythons from {} on the system. Skipping.'.format(versions))
+
+
 def skip_unless_pythons(*versions):
```

Comment:
Note I really want to rename all the functions in this file to `skip_unless_*_present`. All the docstrings already use the word "present", and I think adding it to the name would make it much clearer in calling files what the decorator is doing. Avoided doing this here for scope creep.

Reply:
Done in https://github.com/pantsbuild/pants/pull/7366!